### PR TITLE
HAAR-601: downgrade MFA error

### PR DESCRIPTION
**Pull request by AlTurner-MOJ**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/MfaServiceBasedController.kt [hunk 1]
```diff
@@ -63,8 +63,16 @@ class MfaServiceBasedController(
     @RequestParam token: String?,
     @RequestParam mfaPreference: MfaPreferenceType?,
     @RequestParam user_oauth_approval: String?,
-    @ModelAttribute("authorizationRequest") authorizationRequest: AuthorizationRequest,
-  ): ModelAndView = mfaChallengeRequest(error, token, mfaPreference, extraModel(user_oauth_approval, authorizationRequest))
+    @ModelAttribute("authorizationRequest") authorizationRequest: AuthorizationRequest?,
+  ): ModelAndView {
+
+    if (authorizationRequest == null) {
+      telemetryClient.trackEvent("MFAServiceMFAChallengeMissingAuthorizationRequest", null, null)
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/MfaServiceBasedController.kt:70]: Nice and snappy name!

Comment (AlTurner-MOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/MfaServiceBasedController.kt:70]: Haha, I wasn't sure if there was a naming convention in place for these i.e. meaning in this case `MFAServiceMFAChallenge` is a necessary context prefix?

Comment (AlTurner-MOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/MfaServiceBasedController.kt]: Size reduced

**Review by karenmillermoj — APPROVED** [review]
lgtm

**Pull request merged**